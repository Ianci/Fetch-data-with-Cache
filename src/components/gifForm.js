import React, { useState } from 'react';
import useCustomFetch from '../hooks/CustomFetch'
import {useFetchReducer} from '../hooks/CustomFetchReducer'
const GifForm = () => {
    const [ search, setSearch ] = useState('')
    const key = '9EmOFaV2f4ckc3N7qTE7gOYZMx8m7bQ7'
    const url = search && `https://api.giphy.com/v1/gifs/search?api_key=${key}&q=${encodeURI(search)}&limit=10&offset=0&rating=g&lang=en`
    
    //Tanto useFetchReducer como useCustomFetch funcionan sin tocar ninguna otra linea de cÃ³digo
    
    const { loading, data } = useFetchReducer(url, 1250)

    console.log(data)

    const handleChange = e => {
        setSearch(e.target.value)
    }
    return ( 
        <>
        <input type="text" 
        placeholder="Search" 
        onChange={handleChange}
        value={search}
        />
        <div className="container">
        {loading && <p>Loading...</p>}
        {data &&
        (
            <div>
               
            </div>
        )}
        </div>
        </>
     );
}
 
export default GifForm;