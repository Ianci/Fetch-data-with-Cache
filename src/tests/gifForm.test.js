const { shallow } = require("enzyme")
import  gifForm  from '../components/gifForm'
import React from 'react'

describe('Pruebas en el componente <gifForm>', () => {
    
    test('Debe mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<gifForm />)
        expect ( wrapper ).toMatchSnapshot()
    })
    
})
