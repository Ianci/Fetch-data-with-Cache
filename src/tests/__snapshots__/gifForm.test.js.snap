// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <gifForm> Debe mostrar el componente correctamente 1`] = `<gifForm />`;
